PROJ-158 - Sprint 4 conclusion (#91)
SW805F20/report: sections/sprint-4/conclusion.tex

This section concludes sprint 4 and will summarize the progress that has been made on the game and discuss the retrospective meeting conducted at the end of sprint 4.

\subsection{Current product}
The two current states of the game can be seen in \autoref{fig:sprint-4-state-of-game}.
We are currently able to connect to the host in the lobby, as seen on \autoref{fig:sprint-4-lobby}, which will lead to \autoref{fig:sprint-4-game} when the host has started the game.
\begin{figure}[H]
    \centering
    \begin{subfigure}{.5\textwidth}
        \centering
        \includegraphics[width=1\linewidth]{figures/sprint-4-lobby.PNG}
        \caption{The lobby where you need to input the IP address to connect.}
        \label{fig:sprint-4-lobby}
    \end{subfigure}
    \begin{subfigure}{.4\textwidth}
        \centering
        \includegraphics[width=.8\linewidth]{figures/sprint-4-game.PNG}
        \caption{The current playing field with goal zones, goal score and players.}
        \label{fig:sprint-4-game}
    \end{subfigure}
    \caption{The current state of the game}
    \label{fig:sprint-4-state-of-game}
\end{figure}

\subsubsection{Pozyx}

> There are not a lot of tasks missing with Pozyx, but there some bugs that needs to get fixed within the next sprint.

Reviewer: ```suggestion
There are not many Pozyx programming tasks left, but there are some bugs that needs to be fixed during the next sprint.
```

\\
During the initial test of the program described in \autoref{sec:initial-test} several exceptions occurred, which were caused by the input data not being correctly formatted.

> This should be fixed in the following sprint.

Reviewer: ```suggestion
```

Reviewer: eller flyt ned under næste bug du nævner

Problems with the firmware were also discovered in the initial test, where Pozyx would not send positional data to the host, but only coordinates of $0.0$.

\subsubsection{Game}

> If the problems with Pozyx gets fixed, the game should have implemented the majority of the requirements defined for the MVP in \autoref{sec:MVP}.

Reviewer 2: ```suggestion
If the problems with Pozyx get fixed, the game should have implemented the majority of the requirements defined for the MVP in \autoref{sec:MVP}.
```

However, in the current iteration of the game, the teams are unable to win the game.

> This is a highly prioritized task on Unity, which we intend to implement in the last sprint.

Reviewer: ```suggestion
This is a highly prioritized Unity task, which we intend to implement in the last sprint.
```

\subsubsection{Networking}
To be able to win the game, it is necessary for the host to be able to send a message that the game has ended to the clients.

> Therefore a package needs to be specified for this purpose to be implemented along with the ability to win a game.

Reviewer 2: skriv måske noget med at det også indebærer at hosten definerer en mængde mål for at vinde og sp check på det når der mål i guess, så er der lidt mere 🤔

\subsection{Retrospective on the process}
This subsection will elaborate on the retrospective meeting that was held on the May 12th.

\subsubsection*{How does the process with Jira work?}
The process with Jira will change with the last sprint.
For previous sprints there was a \textit{Suggested} column and a \textit{Discussed} column, but in the upcoming final sprint the \textit{Discussed} column will no longer be used.
Tasks will either be moved to \textit{Chosen for development} or a \textit{Future work} column that was added, rather than go through the \textit{Discussed} column.
The next sprint will be the last, and therefore it is necessary to prioritize it differently.
Either the task has to be completed, or it has to be documented in \textit{Future work}.

\subsubsection*{How has pair programming been working?}

> Pair programming works well, but there are some challenges with online pair programming in Unity.

Reviewer: Er alt det tidligere ikke skrevet i datid??

\section{Sprint 4 conclusion}\label{sec:sprint4conclusion}
This section concludes sprint 4 and will summarize the progress that has been made on the game and discuss the retrospective meeting conducted at the end of sprint 4.

\subsection{Current product}

> In \autoref{fig:sprint-4-state-of-game} the two current states of the game can be seen.

Reviewer 2: ```suggestion
The two current states of the game can be seen in \autoref{fig:sprint-4-state-of-game} .
```

We are currently able to connect to the host in the lobby, as seen on \autoref{fig:sprint-4-lobby}, which will lead to \autoref{fig:sprint-4-game} when the host has started the game.
\begin{figure}[H]
    \centering
    \begin{subfigure}{.5\textwidth}
        \centering
        \includegraphics[width=1\linewidth]{figures/sprint-4-lobby.PNG}
        \caption{The lobby where you need to input the IP address to connect.}
        \label{fig:sprint-4-lobby}
    \end{subfigure}
    \begin{subfigure}{.4\textwidth}
        \centering
        \includegraphics[width=.8\linewidth]{figures/sprint-4-game.PNG}
        \caption{The current playing field with goal zones, goal score and players.}
        \label{fig:sprint-4-game}
    \end{subfigure}
    \caption{The current state of the game}
    \label{fig:sprint-4-state-of-game}
\end{figure}

\subsubsection{Pozyx}

> There are not a lot of tasks missing with Pozyx, but there are several tasks that needs to be completed in the next sprint to make it work.

Reviewer 2: ?? Der er ikke mange der mangler, men der er mange der skal laves

\\\\

> During the initial test of the program describe in \autoref{sec:initial-test} several exception occurred, which was because that the input data was not correctly formatted.

Reviewer 2: ```suggestion
During the initial test of the program described in \autoref{sec:initial-test} several exceptions occurred, which were caused by the input data not being correctly formatted.
```

> This should be easy to fix.

Reviewer 2: ```suggestion
This should be fixed in the following sprint.
```

> In the initial test there were also discovered problems with the firmware, where Pozyx would not send positional data to the host, but only send coordinates of 0.0.

Reviewer 2: ```suggestion
Problems with the firmware were also discovered in the initial test, where Pozyx would not send positional data to the host, but only coordinates of $0.0$.
```

\subsubsection{Game}

> If the problems with Pozyx gets fixed, the game should be working fine.

Reviewer 2: ```suggestion
If the problems with Pozyx get fixed, the game should have implemented the majority of the requirements defined for the MVP.
```

Reviewer 2: add ref til mvp btw

> But in the current iteration of the game, you are unable to win the game.

Reviewer 2: ```suggestion
However, in the current iteration of the game, the teams are unable to win the game.
```

This is a highly prioritized task on Unity, which we intend to implement in the last sprint.

> \subsubsection{Networking}

Reviewer 2: ret pepega subsubsection det her

Reviewer 2: har tilføjet en sætning
ved ikke rigtigt om vi kan finde på lidt mere 🤔

> To be able to win the game, it is necessary for the host signal winning the game to the clients and therefore a package needs to be specified.

Reviewer 2: ```suggestion
To be able to win the game, it is necessary for the host to be able to send a message that the game has ended to the clients.
Therefore a package needs to be specified for this purpose to be implemented along with the ability to win a game.
```

\subsection{Retrospective on the process}

> This subsection will elaborate the retrospective meeting that was held on the May 12th.

Reviewer 2: ```suggestion
This subsection will elaborate on the retrospective meeting that was held on the May 12th.
```

\subsubsection*{How does the process with Jira work?}
The process with Jira will change with the last sprint.

> For previous sprints there were a \textit{Suggested} column and \textit{Discussed} column, but in the upcoming sprint the \textit{Discussed} column will no longer be used.

Reviewer 2: ```suggestion
For previous sprints there was a \textit{Suggested} column and a \textit{Discussed} column, but in the upcoming final sprint the \textit{Discussed} column will no longer be used.
```

> It will either be moved to \textit{Chosen for development} or a \textit{Future work} column that was added.

Reviewer 2: ```suggestion
Tasks will either be moved to \textit{Chosen for development} or a \textit{Future work} column that was added, rather than go through the \textit{Discussed] column.
```

> This sprint will be the last, and therefore we find it less important to prioritize the tasks.

Reviewer 2: vi prioriterer vel stadig i form af hvad vi ligger over i future work? tror ikke rigtigt den her formulering fungerer

Either the task has to be completed, or it has to be documented in \textit{Future work}.

\subsubsection*{How has pair programming been working?}

> Pair programming works well, but there are some challenges with online pair programming in Unity, as it is required to compile to an android device, which is more difficult to share with the other programmer.

Reviewer 2: ```suggestion
Pair programming works well, but there are some challenges with online pair programming in Unity.
The main challenges is that, when testing, it is required that the game is compiled to an android device, which is more difficult to share with the other programmer, unless they both compile the game.
```

\subsubsection*{How is the daily stand-up working?}

> When the members of the group does not have a task, the daily stand up seems pointless, as they do not have much to say.

Reviewer 2: ```suggestion
When the members of the group do not have a task, the daily stand up seems pointless, as they do not have much to say.
```

Reviewer 2: det der med at skrive at nogle ikke har en task skal måske ændres til ngoet med at personen færdiggør en opgave men ikke får en ny en før dagen efter

> To change this by the start of the day people will have 5 minutes to look at tasks and pick one, so that they have the opportunity to discuss that.

Reviewer 2: ```suggestion
To change this, at the start of the day the developers should have five minutes to look at possible tasks for the day and pick one, so that they have the opportunity to discuss that during the stand-up meeting.
```

\\\\

> There were also a comment that estimations of completed tasks were inaccurate.

Reviewer 2: ```suggestion
There was also a comment that estimations of completed tasks were inaccurate.
```

> So whenever someone said the previous day that they would have completed the task, they have to specify the reasons why they did not complete the tasks.

Reviewer 2: ```suggestion
Whenever someone estimated during a stand-up that they would have a task completed by the end of the day, it would often not be completed in time for the next stand-up.
The developer would then give the same estimation, saying it would now be done by the end of the day, and the pattern could repeat.
In order to remedy this, and attempt to ensure that the estimations could become more accurate, the developers should have to specify the reasons why they did not complete the tasks, if they exceeded the estimated duration of the tasks given in a stand-up meeting.
```

> \\\\

Reviewer 2: ```suggestion
```

> Sometimes it is suspected that the motivation is low, and to help this we will introduce pair report writing.

Reviewer 2: ```suggestion
It is suspected that a possible cause of some of this lazy estimation could be that the motivation for certain tasks might be low, and to help this we will introduce pair report writing for larger report tasks, such that the developer pair can help keep each other accountable.
```

> If people have tried it in the upcoming days, it will be evaluated to see if it is something that is well liked.

Reviewer 2: ```suggestion
```

\subsubsection*{How are the reviews going?}

> The group members are more aware of doing the reviews, so reviews often gets done at the same day or in the morning the next day.

Reviewer 2: ```suggestion
The group members are now more aware of doing the reviews after it was discussed during previous sprint retrospectives, so reviews often get done at the same day or in the morning the next day.
```

Reviewer 2: find lige ud af hvilket sprint retro det bvar diskuteret i og ref til det

\subsubsection*{Closing comments}

> At the end of the sprint retrospective we went through the tasks that was completed in the sprint to see what was interesting to describe in the report.

Reviewer 2: synes det er lidt pepega det her. enten skal der lidt uddybves eller også skal det bare væk

\end{figure}

\subsubsection{Pozyx}
There are not many Pozyx programming tasks left, but there are some bugs that needs to be fixed during the next sprint.
\\
During the initial test of the program described in \autoref{sec:initial-test} several exceptions occurred, which were caused by the input data not being correctly formatted.
This should be fixed in the following sprint.
Problems with the firmware were also discovered in the initial test, where Pozyx would not send positional data to the host, but only coordinates of $0.0$.

\subsubsection{Game}
If the problems with Pozyx get fixed, the game should have implemented the majority of the requirements defined for the MVP in \autoref{sec:mvp}.
However, in the current iteration of the game, the teams are unable to win the game.
This is a highly prioritized Unity task, which we intend to implement in the last sprint.

\subsubsection{Networking}
To be able to win the game, it is necessary for the host to be able to send a message that the game has ended to the clients.
Therefore a package needs to be specified for this purpose to be implemented along with the ability to win a game.

> The host also needs to send a certain amount of goals needed to be scored before winning the game, which requires a package to be updated.

Reviewer: ```suggestion
The host also needs to send information about the amount of goals which a team needs to score before they win the game, which requires the package containing player data and now also goal amount to be updated.
```
